kernel/kernel-api/src/main/java/io/delta/kernel/data/PaginationFilteredColumnarBatch.java

```diff
@@ -0,0 +1,43 @@
+package io.delta.kernel.data;
+
+import static io.delta.kernel.internal.util.Preconditions.checkArgument;
+
+import java.util.Optional;
+
+public class PaginationFilteredColumnarBatch extends FilteredColumnarBatch {
+  private final Optional<String> filePath;
+  private final Optional<Integer> preComputedNumSelectedRows;
+
+  public PaginationFilteredColumnarBatch(
+      ColumnarBatch data,
+      Optional<ColumnVector> selectionVector,
+      String filePath,
```

Comment:
does not need to be optional now, eh? this is an internal class we are producing that is only for paginated log replay